src/PosthogAnalytics.ts

```diff
@@ -334,4 +358,26 @@ export class PosthogAnalytics {
                 this.updateAnonymityFromSettings(!!newValue);
             });
     }
+
+    public setAuthenticationType(authenticationType: Signup["authenticationType"]): void {
+        this.authenticationType = authenticationType;
+        window.sessionStorage.setItem(STORAGE_AUTH_TYPE_KEY, authenticationType);
```

Comment:
Not sure how helpful sessionStorage is here given that `trackNewUserEvent` is only called if `setJustRegisteredUserId()` return true which does not persist reloads (`justRegisteredUserId` is stored in memory only)

So whilst `authenticationType` may correctly persist reboots via sessionStorage, it'll subsequently never be read

Reply:
I guess with that new approach the `sessionStorage` can be dismissed, you're right!